curriculum/challenges/english/25-front-end-development/quiz-javascript-problem-solving-and-algorithmic-thinking/66edd4bc0ce399f475ad6f9b.md

```diff
@@ -17,439 +17,439 @@ Answer all of the questions below correctly to pass the quiz.
 
 #### --distractors--
 
-Placeholder distractor 1
+Sort the array in decreasing order and place zeros at the start.
 
 ---
 
-Placeholder distractor 2
+Sort the array ascending order.
 
 ---
 
-Placeholder distractor 3
+Use recursion to remove all zeros and add them to the end.
 
 #### --answer--
 
-Placeholder answer
+Use a two-pointer technique to swap non-zero elements with zero elements.
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Given a sentence, such as `"I am a frontend engineer"`, what approach can be taken to reverse the order of the words, producing the output `"engineer frontend a am I"`?
 
 #### --distractors--
 
-Placeholder distractor 1
+Split the string into words using `split(" ")`, reverse the array, and join it back into a string using `join("_")`.
 
 ---
 
-Placeholder distractor 2
+Sort the words alphabetically and return the result.
 
 ---
 
-Placeholder distractor 3
+Use regular expressions to reverse each word.
 
 #### --answer--
 
-Placeholder answer
+Split the string into words using `split(" ")`, reverse the array, and join it back into a string using `join(" ")`.
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Given a string, what would be the best approach to find the first non-repeating character? For example, in the string `"codecamp"`, it would be `'o'`.
 
 #### --distractors--
 
-Placeholder distractor 1
+Sort the string and check for consecutive duplicates.
 
 ---
 
-Placeholder distractor 2
+Convert the string to an array, then filter for unique elements.
 
 ---
 
-Placeholder distractor 3
+Use a regular expression to find the first non-repeat.
 
 #### --answer--
 
-Placeholder answer
+Iterate over the string twice, once to count number of occurrences and once to find the first unique character.
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What approach can be used to encode a string by counting consecutive repeated characters? For example, given the input` "mississippi"`, the output should be `"m1i1s2i1s2i1p2i1"`.
 
 #### --distractors--
 
-Placeholder distractor 1
+Use regular expressions to match repeated characters.
 
 ---
 
-Placeholder distractor 2
+Use `map()` to map consecutive characters to the output.
 
 ---
 
-Placeholder distractor 3
+Sort the string and count occurrences.
 
 #### --answer--
 
-Placeholder answer
+Iterate through the string, counting consecutive occurrences of each character while building answer.
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What approach can be used to reverse the digits of an integer? For example, given the input `1234`, the output should be `4321`.
 
 #### --distractors--
 
-Placeholder distractor 1
+Use bitwise operations to reverse the digits.
 
 ---
 
-Placeholder distractor 2
+Convert the integer to a binary number and reverse it.
 
 ---
 
-Placeholder distractor 3
+Use recursion to reverse digits.
```

Comment:
This can also be a solution 🤷

Reply:
Yeah, this can be a solution. Will update it in next commit.